cvat/apps/events/handlers.py

```diff
@@ -373,8 +373,8 @@ def handle_delete(scope, instance, store_in_deletion_cache=False, **kwargs):
         user_email=uemail,
     )
 
-def handle_annotations_change(instance, annotations, action, **kwargs):
-    _annotations = deepcopy(annotations)
+def handle_annotations_change(instance, annotations, action, should_deepcopy=True, **kwargs):
```

Comment:
I think this breaks encapsulation of the function, if a parameter with such name is required to be specified by the client code. I can see several ways to improve it:
- rename to "annotations_mutable" or add a kw-parameter "mutable_annotations"
- infer it automatically inside the function, if the operation is "delete". The problem here is this code is not responsible for this logic. The same problem is with the implemented passing of `should_deepcopy=False` in the `delete()` call - it looks like it actually must be forwarded all the way from the place the class instance is created to be fully correct.

Reply:
I think the function may be just refactored to avoid modifying its arguments

Reply:
So now it still will be copying, but without calling `deepcopy`. Do we really need to send all these annotations, can it be replaced by some summary? Can we modify the recipient side to get only required fields?

Reply:
> So now it still will be copying, but without calling deepcopy

The main difference here is that we're only copying outer layers of the data, while nested fields remain the same objects. I'd say this has negligible impact on memory/performance, especially compared to previous implementation. 

> Do we really need to send all these annotations, can it be replaced by some summary? Can we modify the recipient side to get only required fields?

While I think it certainly worth a discussion, but I don't have enough context to answer these questions yet.